### DISCUSSION TITLE
Dropbox no longer supporting Long lived Tokens

### DISCUSSION THREAD

**Discussion by rbarbier1967** [AzuraCast/AzuraCast#5356] [General]
Below was a reply in their forums, I was searching why tokens were ending/expiring...
Hope The Programming team sees this and finds a solution. I tried S3 compatible object storage to no avail lol Got Dropbox going to find out this.  sigh

"Dropbox is in the process of switching to only issuing short-lived access tokens (and optional refresh tokens) instead of long-lived access tokens. You can find [more information on this migration here](https://dropbox.tech/developers/migrating-app-permissions-and-access-tokens).

Apps can still get long-term access by requesting "offline" access though, in which case the app receives a "refresh token" that can be used to retrieve new short-lived access tokens as needed, without further manual user intervention. You can find more information in the [OAuth Guide](https://developers.dropbox.com/oauth-guide) and [authorization documentation](https://www.dropbox.com/developers/documentation/http/documentation#authorization). You can find an example of using a refresh token to get a new short-lived access token, without manual interaction, here, under ["Example: refresh token request"](https://www.dropbox.com/developers/documentation/http/documentation#oauth2-token).

For reference, while the creation of new long-lived access tokens is now deprecated, we don't currently have a plan to disable existing long-lived access tokens. (If that changes, we will of course announce that ahead of time.) That being the case, you can continue using existing long-lived access token(s) without interruption, if you have any. Also, note though that after the change you won't be able to create new long-lived access tokens."

**Comment by BusterNeece** [comment]
@rbarbier1967 Welp.

Thanks for the update...I'll check with our upstream dependency on Dropbox to see if there's anything we can do about it.
